content/blog/2019-10-23-brilpre.md

```diff
@@ -0,0 +1,221 @@
+| ./test/pre_test/complex_loop.json         | 14         | 15        | 7.1%  | 4009          | 4010         | 0.0%  | 4004               | 3005              | -25.0% |
+| ./test/pre_test/complex_loop2_unsafe.json | 18         | 18        | 0.0%  | 7852          | 7852         | 0.0%  | 6867               | 6867              | 0.0%   |
+| ./test/pre_test/complex_loop2.json        | 19         | 21        | 10.5% | 6007          | 6508         | 8.3%  | 5501               | 5001              | -9.1%  |
+| ./test/pre_test/register_presure.json     | 14         | 16        | 14.3% | 9             | 10           | 11.1% | 4                  | 4                 | 0.0%   |
+| ./test/pre_test/simple_loop.json          | 9          | 13        | 44.4% | 7             | 8            | 14.3% | 3                  | 2                 | -33.3% |
+| ./test/pre_test/logic.json                | 21         | 21        | 0.0%  | 6             | 6            | 0.0%  | 2                  | 2                 | 0.0%   |
+| ./test/pre_test/print.json                | 3          | 3         | 0.0%  | 3             | 3            | 0.0%  | 0                  | 0                 | na     |
+| ./test/pre_test/add.json                  | 3          | 3         | 0.0%  | 3             | 3            | 0.0%  | 1                  | 1                 | 0.0%   |
+| ./test/pre_test/loop_invariant.json       | 11         | 12        | 9.1%  | 400005        | 400006       | 0.0%  | 400000             | 300001            | -25.0% |
+| ./test/pre_test/fibonacci.json            | 17         | 17        | 0.0%  | 648           | 648          | 0.0%  | 403                | 403               | 0.0%   |
+| ./test/pre_test/complex_loop3.json        | 24         | 26        | 8.3%  | 66004         | 66006        | 0.0%  | 63999              | 53001             | -17.2% |
+| ./test/pre_test/gcd.json                  | 16         | 16        | 0.0%  | 141           | 141          | 0.0%  | 92                 | 92                | 0.0%   |
+| ./test/pre_test/factorial.json            | 14         | 14        | 0.0%  | 100008        | 100008       | 0.0%  | 100003             | 100003            | 0.0%   |
+| ./test/df_test/cond.json                  | 15         | 15        | 0.0%  | 9             | 9            | 0.0%  | 3                  | 3                 | 0.0%   |
+| ./test/df_test/fact.json                  | 13         | 13        | 0.0%  | 62            | 62           | 0.0%  | 42                 | 42                | 0.0%   |
+
+Above is a table showing all the results, 
+unfortunately for most cases, the improvement is not significant, 
+part of the reasons should be that most programs are short and do not involve loops.
+But for all programs with loops that I manually designed,
+this tool can successfully detect them, generate correct new code,
+and provide a significant performance improvement.
```

Comment:
It is cool that you ran your optimization on such a large number of programs! As you know, most of these were written as tiny tests to check the correctness of individual transformations, so they're not really realistic code in any sense. So I think it's fine that there were no optimization opportunities for most of them.

Reply:
Glad that you think it is cool!